### ISSUE TITLE
Unable to generate l2 attack examples

### ISSUE THREAD

**Issue by ericcheng09** [carlini/nn_robust_attacks#21]
I have a pre-trained keras MNIST model and I want to use my model to generate adversarial examples.

Here is the code I change in class MNISTModel() in setup_mnist.py :
` 
      
        self.num_channels = 1
        self.image_size = 28
        self.num_labels = 10
        K.set_session(session)
        model = load_model('MNIST_model1.h5')
        self.model = model
`

But when I ran the code, I got some examples that have all zeros.

Should I load the model like this?
Thank you

**Comment by loser27** [comment]
Yes, I am also facing a similar situation. Did you find some workaround to this ericcheng09.
Loading the model trained using train_models.py also some outputs adversarial examples which are having all zeros.

Any suggestions?
Thanks

**Comment by carlini** [comment]
When it outputs all zeros, that's an indication it failed to construct an adversarial example. Most of the time, this means the keras model you are loading includes a softmax activation layer as the final layer, and you should remove this (or otherwise change the predict function to be tf.log(self.model(x)).

**Comment by loser27** [comment]
Thanks for the feedback. It works for most the examples now.

**Comment by kakacharles10** [comment]
Can I generate cifar-10 attack examples on Windows?

**Comment by carlini** [comment]
I've never tested that.
